rpc/lib/client/http_client.go

```diff
@@ -83,11 +83,19 @@ func makeHTTPClient(remoteAddr string) (string, *http.Client) {
 
 //------------------------------------------------------------------------------------
 
+// jsonRPCBufferedRequest allows us to track buffered requests, as well as their
+// anticipated response structures.
+type jsonRPCBufferedRequest struct {
+	request types.RPCRequest
+	result  interface{}
+}
+
 // JSONRPCClient takes params as a slice
 type JSONRPCClient struct {
-	address string
-	client  *http.Client
-	cdc     *amino.Codec
+	address  string
+	client   *http.Client
+	cdc      *amino.Codec
+	reqBatch []jsonRPCBufferedRequest
```

Comment:
shouldn't we initialize it in NewJSONRPCClient?

Reply:
Yes, I can't understand how this wasn't caught by the tests... Very strange. Fixed now - thanks! 👍